### PR TITLE
Cast solver exit condition to signed int

### PR DESCRIPTION
**Pull request by shueja**

### CODE REVIEW

File: trajoptlib/src/RustFFI.hpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ static void trycatch(Try&& func, Fail&& fail) noexcept try {
   fail(e.what());
 } catch (const sleipnir::SolverExitCondition& e) {
   // TODO: Use std::to_underlying() from C++23
-  fail(std::to_string(static_cast<uint8_t>(e)));
+  fail(std::to_string(static_cast<int8_t>(e)));
```
Comment (calcmogul) [inline trajoptlib/src/RustFFI.hpp:24]: Also add `#include <type_traits>`.
```suggestion
  fail(std::to_string(static_cast<std::underlying_type_t<decltype(e)>>(e)));
```

**Review by oh-yes-0-fps — APPROVED** [review]
Happens to the best of us

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: trajoptlib/src/RustFFI.hpp (hunk 1 above)
Comment (shueja) [reply trajoptlib/src/RustFFI.hpp:24]: Can we do that now? I thought we didn't support c++23 fully yet

Comment (calcmogul) [reply trajoptlib/src/RustFFI.hpp:24]: Reread the type more carefully.

**Review by calcmogul — APPROVED** [review]

File: trajoptlib/src/RustFFI.hpp [hunk 2]
```diff
@@ -21,7 +22,8 @@ static void trycatch(Try&& func, Fail&& fail) noexcept try {
   fail(e.what());
 } catch (const sleipnir::SolverExitCondition& e) {
   // TODO: Use std::to_underlying() from C++23
-  fail(std::to_string(static_cast<uint8_t>(e)));
+  // numerical value of the enum, converted to string
+  fail(std::to_string(static_cast<std::underlying_type_t<decltype(e)>>(e)));
```
Comment (calcmogul) [inline trajoptlib/src/RustFFI.hpp:26]:
```suggestion
  fail(std::to_string(
      static_cast<std::underlying_type_t<sleipnir::SolverExitCondition>>(e)));
```

**Review by calcmogul — CHANGES_REQUESTED** [review]

**Review by calcmogul — APPROVED** [review]

**Pull request merged**